src/server/detail/snapshot_storage.h

```diff
@@ -0,0 +1,110 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+
+#pragma once
+
+#include <filesystem>
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "io/io.h"
+#include "server/common.h"
+#include "util/cloud/aws.h"
+#include "util/fibers/fiberqueue_threadpool.h"
+#include "util/uring/uring_file.h"
+
+namespace dfly {
+namespace detail {
+
+namespace fs = std::filesystem;
+
+using namespace std::literals;
+
+constexpr std::string_view kS3Prefix = "s3://"sv;
+
+const size_t kBucketConnectMs = 2000;
+
+enum FileType : uint8_t {
```

Comment:
storage can be both IO_URING and DIRECT

Reply:
File type is a bitmask (see https://github.com/dragonflydb/dragonfly/pull/1827#discussion_r1319981059) - which seemed the easiest way to combine options